### DISCUSSION TITLE
які є антіпаттерни асинхронного програмування

### DISCUSSION THREAD

**Discussion by SzymonKovalski** [HowProgrammingWorks/LiveQA#11] [Q&A]
я не можу найти списку

**Comment by tshemsedinov** [comment]
- Не ігнорувати помилки у `callback` та `Promise`, завжди обробляти `.on('error', ...)`
- Дуже ретельно робити ескалацію помилок між абстракціями різного рівня та різними асинхронными контрактами
- Не використовувати `EventEmitter` у якості GOTO, тобто для управління потоком виконання (у сенсі послідовності), тільки обробляти події, як передбачено семантикою цієї абстракції
- Не повертати Promise з асинхронних функцій (додатково не загортати у `return new Promise`)

Відповідь у відео: https://youtu.be/DFBm4cNYaAY
Але потрібна окрема лекція по цьому питанию.
